tests/functional/external_mode/test_virtualization_storageclass.py

```
@@ -86,52 +86,128 @@ def setup_and_teardown_cnv(self):
+                except CommandFailed as e:
+                    # Specific scenario handling: catch expected cluster API failures on resource dropping
+                    if primary_exception_active:
+                        log.warning(
+                            "VM workload deletion failed during teardown, but suppressing "
+                            "to prioritize the primary test failure. Cleanup error: %s",
+                            str(e),
+                        )
+                    else:
+                        log.error(
+                            "VM workload deletion resource trace failed: %s", str(e)
+                        )
```

Ruff (0.15.20)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)